commands/utils/SourceArchiveUtility.ts

```diff
@@ -0,0 +1,114 @@
+import { ContainerRegistryManagementClient } from 'azure-arm-containerregistry/lib/containerRegistryManagementClient';
+import { DockerBuildRequest, FileTaskRunRequest, Registry } from 'azure-arm-containerregistry/lib/models';
+import { Subscription } from 'azure-arm-resource/lib/subscription/models';
+import { BlobService, createBlobServiceWithSas } from "azure-storage";
+import * as fse from 'fs-extra';
+import * as os from 'os';
+import * as tar from 'tar';
+import * as url from 'url';
+import vscode = require('vscode');
+import { IActionContext, IAzureQuickPickItem } from 'vscode-azureextensionui';
+import { ext } from '../../extensionVariables';
+import { getBlobInfo, getResourceGroupName, streamLogs } from "../../utils/Azure/acrTools";
+import { AzureUtilityManager } from '../../utils/azureUtilityManager';
+import { Item } from '../build-image';
+import { quickPickACRRegistry, quickPickSubscription } from './quick-pick-azure';
+import { quickPickDockerFileItem, quickPickImageName, quickPickYamlFileItem } from './quick-pick-image';
+import { quickPickWorkspaceFolder } from './quickPickWorkspaceFolder';
+
+const idPrecision = 6;
+const vcsIgnoreList = ['.git', '.gitignore', '.bzr', 'bzrignore', '.hg', '.hgignore', '.svn']
+const status1 = vscode.window.createOutputChannel('Scheduled ACR Run');
+
+export type runRequestType =
```

Comment:
Enum would be better

Reply:
I actually was using enum before. I'll change it, altho why is enum better? If you don't mind.
Thanks

Reply:
Not 100% sure it's better, but it seems better typed if there's no reason for it to be a string.  Although I believe enum values can be strings.

Reply:
I'm using enum as a string, should I use numbers instead? 
For me, it was just a little help to make sure that the person using the function understands those are the option available because there are more types of run requests.

Reply:
You're using a string literal union type.  But I guess I'm not sure enums are better, so you can leave it alone.